openedx/core/djangoapps/geoinfo/middleware.py

```diff
@@ -33,14 +33,18 @@ def process_request(self, request):
         new_ip_address = get_real_ip(request)
         old_ip_address = request.session.get('ip_address', None)
 
+        reader = geoip2.database.Reader(settings.GEOIP_PATH)
         if not new_ip_address and old_ip_address:
             del request.session['ip_address']
             del request.session['country_code']
         elif new_ip_address != old_ip_address:
-            if new_ip_address.find(':') >= 0:
-                country_code = pygeoip.GeoIP(settings.GEOIPV6_PATH).country_code_by_addr(new_ip_address)
-            else:
-                country_code = pygeoip.GeoIP(settings.GEOIP_PATH).country_code_by_addr(new_ip_address)
+            try:
+                response = reader.country(new_ip_address)
+                country_code = response.country.iso_code
+            except geoip2.errors.AddressNotFoundError:
+                country_code = "unknown"
+
             request.session['country_code'] = country_code
             request.session['ip_address'] = new_ip_address
             log.debug(u'Country code for IP: %s is set to %s', new_ip_address, country_code)
+        reader.close()
```

Comment:
Same here, close reader inside try block.

Reply:
If the creation of the reader is moved into the elif clause, then this should be as well.   (But again I don't agree that it should be inside the try.)

Reply:
Not understanding the last comment.   The close() should be called event if there is an exception.    But I think we should move the reader creation inside the elif clause, as well as the reader.close(), because if the elif condition is not met, the reader is not used and shouldn't have wasted time doing its initialization.